### PR TITLE
Make final API review changes to file enumeration

### PR DESCRIPTION
**Pull request by JeremyKuhne**
Tweak stack array initialization to just zero the first element in matcher algorithm.

cc: @danmosemsft @terrajobst

### CODE REVIEW

File: src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs [hunk 1]
```diff
@@ -129,6 +129,7 @@ public FileAttributes Attributes
         public DateTimeOffset LastAccessTimeUtc => _status.GetLastAccessTime(FullPath, continueOnError: true);
         public DateTimeOffset LastWriteTimeUtc => _status.GetLastWriteTime(FullPath, continueOnError: true);
         public bool IsDirectory => _status.InitiallyDirectory;
+        public bool IsHidden => _directoryEntry.Name[0] == '.';
```
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs]: What if Name is null or empty?  I see us doing checks elsewhere, like:
https://github.com/dotnet/corefx/pull/27318/files#diff-61f827c8c50fc3e07c9e00ab7bc18bf3R74
for whether the Length == 0 and like:
https://github.com/dotnet/corefx/pull/27318/files#diff-61f827c8c50fc3e07c9e00ab7bc18bf3R91
for whether Name == null.

File: src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemName.cs [hunk 2]
```diff
@@ -141,7 +141,7 @@ private static bool MatchPattern(ReadOnlySpan<char> expression, ReadOnlySpan<cha
             Span<int> temp = stackalloc int[0];
             Span<int> currentMatches = stackalloc int[16];
             Span<int> priorMatches = stackalloc int[16];
-            priorMatches.Clear();
+            priorMatches[0] = 0;
```
Comment (stephentoub) [inline src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemName.cs]: Thanks.

File: src/System.IO.FileSystem/tests/Enumeration/AttributeTests.netcoreapp.cs [hunk 3]
```diff
@@ -111,5 +112,30 @@ public void DirectoryAttributesAreExpected()
...
+            fileTwo.Create().Dispose();
+            if (PlatformDetection.IsWindows)
+                fileTwo.Attributes = fileTwo.Attributes | FileAttributes.Hidden;
+
+            IEnumerable<string> enumerable = new FileSystemEnumerable<string>(
+                testDirectory.FullName,
+                (ref FileSystemEntry entry) => entry.ToFullPath(),
+                new EnumerationOptions() { AttributesToSkip = 0 })
+            {
+                ShouldIncludePredicate = (ref FileSystemEntry entry) => entry.IsHidden
```
Comment (stephentoub) [inline src/System.IO.FileSystem/tests/Enumeration/AttributeTests.netcoreapp.cs]: `new FileSystemEntry().IsHidden` will null ref.  Do we care?

File: src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs (hunk 1 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem/src/System/IO/Enumeration/FileSystemEntry.Unix.cs]: Good catch- totally missed my doubling that up

File: src/System.IO.FileSystem/tests/Enumeration/AttributeTests.netcoreapp.cs (hunk 3 above)
Comment (JeremyKuhne) [reply src/System.IO.FileSystem/tests/Enumeration/AttributeTests.netcoreapp.cs]: We don't- you can't populate these yourself and we want to discourage newing these up

**Comment by JeremyKuhne** [comment]
Grr. Trying to fix the merge conflict.

**Pull request merged**